src/include/platform/internal/GenericPlatformManagerImpl_FreeRTOS.ipp

```diff
@@ -278,9 +278,9 @@ void GenericPlatformManagerImpl_FreeRTOS<ImplClass>::EventLoopTaskMain(void * ar
 {
     ChipLogDetail(DeviceLayer, "CHIP event task running");
     static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)->Impl()->RunEventLoop();
-    // TODO: At this point, should we not
-    // vTaskDelete(static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)->mEventLoopTask)?
-    // Or somehow get our caller to do it once this thread is joined?
+    vTaskDelete(NULL);
+    vQueueDelete(static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)->mChipEventQueue);
```

Comment:
Seems like we should store `static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)` in a local variable, instead of copy/pasting that mess several times.